openmc/mgxs/library.py

```diff
@@ -819,20 +820,22 @@ def get_xsdata(self, domain, xsdata_name, nuclide='total', xs_type='macro',
             name += '_' + nuclide
         name += '.' + xs_id
         xsdata = openmc.XSdata(name, self.energy_groups)
-        if order is 0:
-            xsdata.order = order
+
+        if order is None:
+            # Set the order to the Library's order (the defualt behavior)
+            xsdata.order = self.legendre_order
         else:
-            msg = 'Generating anisotropic scattering from openmc.Library' \
-                  'objects has not yet been implemented.'
-            raise NotImplementedError(msg)
+            # Set the order of the xsdata object to the minimum of
+            # the provided order or the Library's order.
+            xsdata.order = min(order, self.legendre_order)
```

Comment:
I've got to look at the docstring again (with all these comments, internet explorer is having a fit so I am not going to subject it to loading another page with the source on it right now...), but this behavior in this code (line 830) is the intended behavior.  I chose not to have a warning since the user had to choose to set order to not be equal to the Mgxs or Library order, and therefore, they don't need a warning to tell them that.

For perspective, this whole functionality exists so that one could apply different expansion orders for moderator than fuel, for example (moderator may need P5, but fuel is fine with P0 or P1), so this gives a little to the user to do this in an easy way.  
Note that I have never seen a paper saying this should be done, but we talked about doing it in MPACT and I always thought it a good idea, but putting the machinery in place there was too difficult.  Now its in place here since I kind of wanted to be able to do that from the start.  Of course, the OpenMC MG runtime doesn't depend heavily on scattering order (since the default option is to expand the legendre's in to a tabular distribution) so there isn't as strong of a driver to do that here as there is in an Sn or MOC code, but it allows one to test it out in advance.